packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -142,14 +147,19 @@ - (void)launchUIImagePickerWithSource:(int)imageSource {
   imagePickerController.delegate = self;
   imagePickerController.mediaTypes = @[ (NSString *)kUTTypeImage ];
 
+  BOOL usePHAsset = [[_arguments objectForKey:@"requestFullMetadata"] boolValue];
   self.maxImagesAllowed = 1;
 
   switch (imageSource) {
     case SOURCE_CAMERA:
       [self checkCameraAuthorizationWithImagePicker:imagePickerController];
       break;
     case SOURCE_GALLERY:
-      [self checkPhotoAuthorizationWithImagePicker:imagePickerController];
+      if (usePHAsset) {
+        [self checkPhotoAuthorizationWithImagePicker:imagePickerController];
```

Comment:
According to the thread you linked to, this change is only correct on iOS 11+, and iOS 9 and 10 still need permissions.